### ISSUE TITLE
Jenkins: Green, even if there are failures in All Plugins tests

### ISSUE THREAD

**Issue by suyograo** [elastic/logstash#2770]
http://build-eu-00.elasticsearch.org/view/LS%201.5/job/Logstash_1.5_All_Plugins_test/222/console

```
Finished in 1 minute 24 seconds
604 examples, 6 failures

Failed examples:
```

**Comment by ph** [comment]
gist of the build log.
https://gist.github.com/ph/398289bb8a81e837c0ef

**Comment by suyograo** [comment]
Another random failure http://build-eu-00.elasticsearch.org/view/LS%201.5/job/Logstash_1.5_All_Plugins_test/224

**Comment by mrsolo** [comment]
Jenkins uses the return shell script return code to set the build status.  

https://wiki.jenkins-ci.org/display/JENKINS/Text-finder+Plugin is useful to address this issue

**Comment by purbon** [comment]
@mrsolo @suyograo Yap, makes sense! my fault here, will fix the script.

**Comment by purbon** [comment]
I guess however that the wrong error code gets captured by the coveralls calls.

**Comment by purbon** [comment]
Test are reporting always with exit_code 0, even if they had failures, ... we see false positive also in core, as you can see in https://gist.github.com/jsvd/a0f98de236ca493ca196

**Comment by purbon** [comment]
Closed by #2920
